### PR TITLE
feature/EXLM-175

### PR DESCRIPTION
**Pull request by adobenitesh**
Jira ID: EXLM-175

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live/docs/authoring-guide-exl/using/markdown/cheatsheet
- After: https://feature-EXLM-175--exlm--adobe-experience-league.hlx.live/docs/authoring-guide-exl/using/markdown/cheatsheet

### CODE REVIEW

File: scripts/data-service/profile.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+import { signOut } from '../auth/auth-operations.js';
+import loadJWT from '../auth/jwt.js';
+import csrf from '../auth/csrf.js';
+import { JWTTokenUrl } from '../urls.js';
+import { JWT, Organization, Profile, ProfileAttributes } from '../auth/session-keys.js';
+import { request, headerKeys, headerValues } from '../request.js';
+
+// eslint-disable-next-line
+export const lang = document.querySelector('html').lang;
+export const origin = 'https://experienceleague.adobe.com';
```
Comment (Sivaramvt) [inline scripts/data-service/profile.js:11]: Move this API endpoint details in `.../urls.js`

File: scripts/data-service/profile.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        [headerKeys.auth]: sessionStorage.getItem(JWT),
+        [headerKeys.accept]: headerValues.json,
+      },
+      method: 'OPTIONS',
+    });
+
+    if (res.ok) {
+      const data = await res.json();
+
+      sessionStorage.setItem(ProfileAttributes, JSON.stringify(data.data));
```
Comment (Sivaramvt) [inline scripts/data-service/profile.js:92]: Ensure to cleanup on sign out

File: blocks/doc-actions/doc-actions.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  );
+
+  if (isSignedIn) {
+    block.appendChild(authBookmark);
+    const bookmarkAuthed = block.querySelector('.bookmark.auth');
+    if (bookmarkAuthed) {
+      const bookmarkAuthedToolTipLabel = bookmarkAuthed.querySelector('.exl-tooltip-label');
+      const bookmarkAuthedToolTipIcon = bookmarkAuthed.querySelector('.icon.bookmark-icon');
+      if (id.length === 0) {
+        log('Hooking bookmark failed. No id present.');
```
Comment (Sivaramvt) [inline blocks/doc-actions/doc-actions.js:77]: No logging strategy in place yet.

File: scripts/data-service/profile.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  if (result !== null) {
+    if (!meta || Object.keys(meta).length === 0) {
+      meta = await profileAttributes();
+    }
+
+    // eslint-disable-next-line
+    const keys = ['industryInterests', 'role'],
+      complete = Math.ceil((keys.filter((k) => result[k].length > 0).length / keys.length) * 100);
+
+    sessionStorage.setItem(Ready, complete === 100);
```
Comment (Sivaramvt) [inline scripts/data-service/profile.js:158]: Let's not cache unnecessary items

File: scripts/data-service/profile.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    }
+  } else {
+    data[key] = val;
+  }
+
+  const res = await request(profileUrl, {
+    method: 'PATCH',
+    credentials: 'include',
+    headers: {
+      authorization: sessionStorage.getItem(JWT),
```
Comment (Sivaramvt) [inline scripts/data-service/profile.js:208]: use jwt API

File: scripts/data-service/profile.js (hunk 1 above)
Comment (adobenitesh) [reply scripts/data-service/profile.js]: Done

File: scripts/data-service/profile.js (hunk 2 above)
Comment (adobenitesh) [reply scripts/data-service/profile.js:72]: Done

File: blocks/doc-actions/doc-actions.js (hunk 3 above)
Comment (adobenitesh) [reply blocks/doc-actions/doc-actions.js]: Done

File: scripts/data-service/profile.js (hunk 4 above)
Comment (adobenitesh) [reply scripts/data-service/profile.js]: Done

File: scripts/data-service/profile.js (hunk 5 above)
Comment (adobenitesh) [reply scripts/data-service/profile.js]: Done

File: scripts/auth/session-keys.js [hunk 6]
```diff
@@ -1,2 +1,5 @@
 export const JWT = 'JWT';
 export const COVEO_TOKEN = 'coveoToken';
+export const Organization = 'organization';
+export const Profile = 'profile';
+export const ProfileAttributes = 'attributes';
```
Comment (Sivaramvt) [inline scripts/auth/session-keys.js:5]: @adobenitesh Are we still caching these keys in session storage?

**Review by Sivaramvt — APPROVED** [review]

**Review by Sivaramvt — APPROVED** [review]

File: scripts/auth/session-keys.js (hunk 6 above)
Comment (adobenitesh) [reply scripts/auth/session-keys.js:4]: Done

**Pull request merged**